web_src/js/features/repo-home.js

```diff
@@ -152,12 +155,13 @@ export function initRepoTopicBar() {
   });
 
   $.fn.form.settings.rules.validateTopic = function (_values, regExp) {
-    const $topics = $topicDropdown.children('a.ui.label');
-    const status = !$topics.length || $topics.last()[0].getAttribute('data-value').match(regExp);
-    if (!status) {
-      $topics.last().removeClass('green').addClass('red');
+    const topics = topicDropdown.querySelectorAll(':scope > a.ui.label');
+    const status = !topics.length || (topics[topics.length - 1]).getAttribute('data-value').match(regExp);
+    if (!status && topics[topics.length - 1]) {
+      (topics[topics.length - 1]).classList.remove('green');
+      (topics[topics.length - 1]).classList.add('red');
     }
-    return status && !$topicDropdown.children('a.ui.label.red').length;
+    return status && !topicDropdown.querySelectorAll(':scope > a.ui.label.red').length;
```

Comment:
Since the `red` class has been added to the invalid labels:

```suggestion
    return topicDropdown.querySelectorAll('a.ui.label.red').length === 0;
```

"no red label" means the topics are all valid.